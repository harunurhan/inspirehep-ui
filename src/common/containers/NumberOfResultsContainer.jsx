import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

class NumberOfResultsContainer extends Component {
  render() {
    const { numberOfResults } = this.props;
    return <span>{numberOfResults}</span>;
  }
}

NumberOfResultsContainer.propTypes = {
  numberOfResults: PropTypes.number.isRequired,
};

const stateToProps = state => ({
  numberOfResults: state.search.get('total'),
});

export default connect(stateToProps)(NumberOfResultsContainer);
