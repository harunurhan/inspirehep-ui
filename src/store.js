import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import createHistory from 'history/createBrowserHistory';
import { routerMiddleware } from 'react-router-redux';
/* eslint-disable import/no-extraneous-dependencies */
import { createLogger } from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
/* eslint-disable import/no-extraneous-dependencies */

import reducers from './reducers';
import http from './common/http';
import queryParamsParserMiddleware from './middlewares/queryParamsParser';
import persistUserStateMiddleware, {
  reHydrateRootStateWithUser,
} from './middlewares/persistUserState';

export const thunkMiddleware = thunk.withExtraArgument(http);

export const history = createHistory();
const reduxRouterMiddleware = routerMiddleware(history);

const getMiddleware = () => {
  if (process.env.NODE_ENV === 'production') {
    return applyMiddleware(
      reduxRouterMiddleware,
      queryParamsParserMiddleware,
      persistUserStateMiddleware,
      thunkMiddleware
    );
  }
  return applyMiddleware(
    reduxRouterMiddleware,
    queryParamsParserMiddleware,
    persistUserStateMiddleware,
    thunkMiddleware,
    createLogger()
  );
};

export const store = createStore(
  reducers,
  reHydrateRootStateWithUser(),
  composeWithDevTools(getMiddleware())
);
