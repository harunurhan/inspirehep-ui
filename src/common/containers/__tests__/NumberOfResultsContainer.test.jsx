import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import { getStoreWithState } from '../../../fixtures/store';
import NumberOfResultsContainer from '../NumberOfResultsContainer';

describe('NumberOfResultsContainer', () => {
  it('renders initial state ', () => {
    const store = getStoreWithState({
      search: fromJS({
        total: 5,
      }),
    });
    const wrapper = shallow(<NumberOfResultsContainer store={store} />).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
