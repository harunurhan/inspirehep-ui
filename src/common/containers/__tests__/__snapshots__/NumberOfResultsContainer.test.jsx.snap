// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberOfResultsContainer renders initial state  1`] = `
<span>
  5
</span>
`;
